weave/type_handlers/Audio/audio.py

```diff
@@ -1,34 +1,216 @@
+                raise ValueError(
+                    f"Invalid file path {path}, file must end in one of: mp3 or wav"
+                )
+            if not os.path.exists(path):
+                raise ValueError(f"File {path} does not exist")
+            self.data = base64.b64encode(open(path, "rb").read()).decode("ascii")
+            self.fmt = fmt_str
+            return
+
+        elif data and fmt:
+            # Case 1: Receive encoded audio data
+            is_encoded = is_base64(data)
+
+            # Case 2: Receive raw audio data
+            if not is_encoded:
+                if isinstance(data, str):
+                    bytes_data = data.encode()
+                    self.data = base64.b64encode(bytes_data).decode("ascii")
+                    self.fmt = fmt
+                    return
+                else:
+                    self.data = base64.b64encode(data).decode("ascii")
+                    self.fmt = fmt
+                    return
+
+    def export(self, path: Union[str, bytes, Path, os.PathLike]) -> None:
+        with open(path, "wb") as f:
+            f.write(base64.b64decode(self.data))
+
+
+def save(
+    obj: Union[wave.Wave_read, Audio], artifact: MemTraceFilesArtifact, name: str
+) -> None:
+    with artifact.writeable_file_path("metadata.json") as metadata_path:
+        obj_module = obj.__module__
+        obj_class = obj.__class__.__name__
+        with open(metadata_path, "w") as f:
+            metadata = {"_type": f"{obj_module}.{obj_class}"}
+            json.dump(metadata, f)
+
+    if isinstance(obj, wave.Wave_read):
+        original_frame_position = obj.tell()
+        obj.rewind()
+        frames = obj.readframes(obj.getnframes())
+        params = obj.getparams()
+        with artifact.writeable_file_path("audio.wav") as fp:
+            with wave.open(fp, "w") as wav_file:
+                # Exclude nframes param, it is often set as the maximum number of frames
+                # which bumps into the 4GB max file size when creating the wave.Wave_write
+                # header on close.
+                wav_file.setframerate(params.framerate)
+                wav_file.setnchannels(params.nchannels)
+                wav_file.setsampwidth(params.sampwidth)
+                wav_file.setcomptype(params.comptype, params.compname)
+                wav_file.writeframes(frames)
+        # Rewind to the original position
+        obj.setpos(original_frame_position)
+    else:
+        with artifact.writeable_file_path(f"audio.{obj.fmt}") as fp:
+            obj.export(fp)
+
+
+def load(artifact: MemTraceFilesArtifact, name: str) -> "wave.Wave_read | Audio":
+    pytype = None
+    if artifact.path_contents.get("metadata.json"):
+        with open(artifact.path("metadata.json")) as f:
+            pytype = json.load(f).get("_type")
+
+    for filename in artifact.path_contents:
+        path = artifact.path(filename)
+        if filename.startswith("audio."):
```

Comment:
The `audio.` magic string appears in a lot of places.  Better to pull it out as a const at the top